docs/source/dataframe-optimizer.rst

```diff
@@ -38,13 +38,25 @@ The optimizations entail the following steps (this list is not complete):
   ``df.groupby(...).apply(...)`` after a merge operation will not shuffle the data
   again if the groupby happens on the merge columns.
 
+  .. figure:: images/optimizer/avoiding-shuffles.svg
+    :align: center
+
+    Performing two subsequent Joins/Merges on the same join-column(s) will avoid shuffling the
+    data again. The optimizer identifies that the partitioning of the DataFrame is already as
+    expected and thus simplifies the operation to a single Shuffle and a trivial merge operation.
```

Comment:
nit: The term "trivial" merge seems very vague to me, maybe "pre-partitioned" merge highlights what's special/trivial about it?
```suggestion
    expected and thus simplifies the operation to a single Shuffle and a pre-partitioned merge operation.
```